pkg/controller/plan/adapter/ovirt/client.go

```diff
@@ -318,3 +347,104 @@ func (r *Client) cacert() []byte {
 	}
 	return nil
 }
+
+func (r Client) Finalize(vmPrecopies map[ref.Ref][]planapi.Precopy) {
+	r.connect()
```

Comment:
Isn't there the same issue here, where since you're launching this in a goroutine that `Run()` could terminate and fire its `defer`, causing the connection to be closed after this starts?